src/commands/ban_commands.py

```diff
@@ -15,30 +15,45 @@ def create_ban_commands(bot: Bot) -> None:
     @discord.app_commands.describe(
         user="User being banned",
         reason="Reason for ban",
+        delete_messages="Type 'Y' to delete messages or 'N' to keep them (default: 'N')",
     )
     async def ban(
         interaction: discord.Interaction,
         user: discord.Member,
         reason: str,
+        delete_messages: str = "N",  # Default to 'N' for no message deletion
     ):
         """Ban the specified user."""
+        # Normalize and validate the input
+        delete_messages = delete_messages.strip().upper()
+        if delete_messages not in ["Y", "N"]:
+            await interaction.response.send_message(
+                "Invalid value for 'delete_messages'. Please enter 'Y' to delete messages or 'N' to keep them.",
+                ephemeral=True,
+            )
+            return
+
+        delete_messages_flag = delete_messages == "Y"  # Convert to boolean
```

Comment:
what about the case where a user inputs `y`?

Reply:
its not case sensitive